app/com/gu/floodgate/contentsource/ContentSourceApi.scala

```diff
@@ -80,6 +88,49 @@ class ContentSourceApi(
     }
   }
 
+  def startBulkReindexer = Action.async(parse.json) { implicit request =>
+    (request.body \ "environments").validate[List[String]].fold(
+      error => jsonError,
+      environments => {
+        val actorList = environments.flatMap(env => bulkJobActorsMap.get(env))
+        val bulkJobResponse = Future.sequence(actorList.map(actor => (actor ? StartBulkReindex).mapTo[BulkReindexRequestResult]))
+        bulkJobResponse.map { resultList =>
+          val failedOrSucceeded = environments.zip(resultList.map(resultObject => resultObject == CanTrigger)).toMap
+          Ok(Json.toJson(failedOrSucceeded))
+        }
+      }
+    )
+  }
+
+  def checkIfInBulkMode = Action.async { implicit request =>
+    val actorList = bulkJobActorsMap.values.toList
+    val actorEnvs = bulkJobActorsMap.keys.toList
+    val bulkJobsActorStatus = Future.sequence(actorList.map(actor => (actor ? IsActorReindexing).mapTo[BulkReindexRequestResult]))
+    bulkJobsActorStatus.map { actorStatusList =>
+      val isReindexingList = actorEnvs.zip(actorStatusList).collect { case (s, status: IsReindexing) => (s, status) }
+      isReindexingList match {
+        case Nil => Ok(s"""{ "IsReindexing": ${false}}""")
+        case x :: xs => {
+          val resultMap = (x :: xs).toMap
+          Ok(s"""{ "IsReindexing": ${true}, "data": ${Json.toJson(resultMap)} } """)
+        }
+      }
+    }
+  }
+
+  def cancelPendingReindex(id: String, environment: String) = Action.async { implicit request =>
+    bulkJobActorsMap.get(environment) match {
+      case Some(actor) => {
+        val futureActorResponse = (actor ? DropPendingJob(id, environment)).mapTo[DropPendingJobResult]
+        futureActorResponse.map {
+          case CancelledPendingJob => Ok(s"""{ "CancelledJob": ${true}, "data": { "id": ${id}, "env": ${environment}}}""")
```

Comment:
Out of curiosity, why are `true` and `false` always surrounded by braces? They are perfectly valid JSON values:
![value](https://user-images.githubusercontent.com/629976/28771405-99c86f2c-75da-11e7-87b0-e84726e84b1b.gif)